packages/shared-influx-pieces/src/lib/AnalyticsListener.ts

```diff
@@ -0,0 +1,48 @@
+import type { Point } from '@influxdata/influxdb-client';
+import { envParseBoolean, envParseString } from '@skyra/env-utilities';
+import { Listener, type PieceContext } from '@skyra/http-framework';
+import { Tags } from './AnalyticsSchema.js';
+
+export abstract class AnalyticsListener extends Listener {
+	public tags: [Tags, string][] = [];
+
+	public constructor(context: PieceContext, options: AnalyticsListener.Options) {
+		super(context, { ...options, enabled: envParseBoolean('INFLUX_ENABLED') });
+	}
+
+	public override onLoad() {
+		void this.initTags();
+		return super.onLoad();
+	}
+
+	public writePoint(point: Point) {
+		return this.container.analytics!.writeApi.writePoint(this.injectTags(point));
+	}
+
+	public writePoints(points: Point[]) {
```

Comment:
Always prefer readonly arrays:
```suggestion
	public writePoints(points: readonly Point[]) {
```